core/src/main/java/hudson/model/Queue.java

```diff
@@ -2459,10 +2482,8 @@ public CauseOfBlockage getCauseOfBlockage() {
                 return CauseOfBlockage.fromMessage(Messages._Queue_BlockedBy(r.getDisplayName()));
             }
 
-            for (QueueTaskDispatcher d : QueueTaskDispatcher.all()) {
-                CauseOfBlockage cause = d.canRun(this);
-                if (cause != null)
-                    return cause;
+            if (causeOfBlockage != null) {
```

Comment:
Fine for me. Probably it could be checked even before calling `getBlockingActivity()` to speedup it even more. Though it needs more feedback from @jenkinsci/code-reviewers

Reply:
Can I move it to getCauseOfBlockage as well? In this case the load from it would be consistent (at least :) ). But I'm not sure about it... From my personal perspective it would be better just remove it :))